src/Handlers/BookingsStateStatusesHandler.php

```diff
@@ -0,0 +1,201 @@
+     * Endpoint for saving statuses.
+     *
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $statusesEndpoint;
+
+    /**
+     * StatusesHandler constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|Traversable|stdClass $statuses         List of statuses key in application.
+     * @param array|Traversable|stdClass $statusesLabels   Map of known statuses labels.
+     * @param string                     $screenOptionsKey Option key name to save screen statuses config.
+     * @param string                     $statusesEndpoint Endpoint for saving statuses.
+     * @param EventManagerInterface      $eventManager     The event manager.
+     * @param EventFactoryInterface      $eventFactory     The event factory.
+     */
+    public function __construct(
+        $statuses,
+        $statusesLabels,
+        $screenOptionsKey,
+        $statusesEndpoint,
+        $eventManager,
+        $eventFactory
+    ) {
+        $this->statuses       = $statuses;
+        $this->statusesLabels = $statusesLabels;
+
+        $this->screenOptionsKey = $screenOptionsKey;
+        $this->statusesEndpoint = $statusesEndpoint;
+
+        $this->_setEventManager($eventManager);
+        $this->_setEventFactory($eventFactory);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        /* @var $event EventInterface */
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $event->setParams(array_merge(
+            $event->getParams(),
+            $this->_getStatusesParams()
+        ));
+    }
+
+    /**
+     * Get status related parameters to attach on event.
+     *
+     * @since [*next-version*]
+     *
+     * @return array
+     */
+    protected function _getStatusesParams()
+    {
+        return [
+            /*
+             * All available booking statuses in application.
+             */
+            'statuses' => $this->_getTranslatedStatuses($this->statuses, $this->statusesLabels),
+
+            /*
+             * List of booking statuses in screen options section that is checked for filtering bookings.
+             */
+            'screenStatuses' => $this->_getScreenStatuses($this->screenOptionsKey, $this->statuses),
+
+            /*
+             * Endpoint for saving booking statuses that will be selected as default for filtering bookings.
+             */
+            'statusesEndpoint' => $this->statusesEndpoint,
+        ];
+    }
+
+    /**
+     * Get all translated statuses.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|Traversable|stdClass $statuses       List of statuses
+     * @param array|Traversable|stdClass $statusesLabels Map of statuses and it's labels
```

Comment:
See [this comment](https://github.com/RebelCode/rcmod-wp-bookings-ui/pull/9/files#r190314479).